site/lib/src/components/pages/changelog/breaking_changes_legend.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright 2025 The Dart Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:jaspr/dom.dart';
+import 'package:jaspr/jaspr.dart';
+
+class BreakingChangesLegend extends StatelessComponent {
+  const BreakingChangesLegend({required this.onClose, super.key});
+
+  final VoidCallback onClose;
+
+  @override
+  Component build(BuildContext context) {
+    return div(classes: 'legend-overlay', [
+      div(classes: 'legend-dialog', [
+        div(classes: 'legend-header', [
+          const h3([.text('About Breaking Changes')]),
+          button(
+            classes: 'close-button',
+            onClick: onClose,
+            [
+              const span(classes: 'material-symbols-outlined', [
+                .text('close'),
+              ]),
+            ],
+          ),
+        ]),
+        const div(classes: 'legend-content', [
+          p([
+            .text(
+              'This page lists all breaking changes and deprecations in '
```

Comment:
Should this introduction be at the top of the page before the index? Then also be updated to cover the expanded range of the page.